src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -744,6 +744,44 @@ public static char SingletonChar(string set)
             !IsSubtraction(set) &&
             (set[SetStartIndex] == LastChar || set[SetStartIndex] + 1 == set[SetStartIndex + 1]);
 
+        /// <summary>Gets whether the set contains nothing other than a single UnicodeCategory (it may be negated).</summary>
+        /// <param name="set">The set to examine.</param>
+        /// <param name="category">The single category if there was one.</param>
+        /// <param name="negated">true if the single category is a not match.</param>
+        /// <returns>true if a single category could be obtained; otherwise, false.</returns>
+        public static bool TryGetSingleUnicodeCategory(string set, out UnicodeCategory category, out bool negated)
+        {
+            if (set[CategoryLengthIndex] == 1 &&
+                set[SetLengthIndex] == 0 &&
+                !IsSubtraction(set))
+            {
+                short c = (short)set[SetStartIndex];
+
+                if (c > 0)
+                {
+                    if (c != SpaceConst)
```

Comment:
BTW I see `private const short SpaceConst = 100;` and comparisons to it but I don't see where anything is set to SpaceConst - what is it and how does it work? Could use a comment where it's defined?

Reply:
This logic is just copied from https://github.com/dotnet/runtime/blob/3f48cb1840bbc1139a9d411758164defb5f9a94c/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs#L1167.

I'm not 100% certain, but I believe the only places these are "used" is in the SpaceClass and NotSpaceClass consts declared below {Not}SpaceConst:
https://github.com/dotnet/runtime/blob/3f48cb1840bbc1139a9d411758164defb5f9a94c/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs#L53-L54

Someone precreated those char class strings with the known value embedded, and then created the consts to be able to refer to that same value.  At least, that's my guess.